Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -2173,9 +2173,22 @@ private void UpdateResourceAvailability(PerformanceCollector.MachinePerfInfo per
             }
 
 #if PLATFORM_OSX
-            Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
-            var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
-            resourceAvailable &= !(result && (pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel));
+            if (!resourceAvailable)
+            {
+                Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
+                var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
+                resourceAvailable = !(result && (pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel));
+
+                if (!result)
+                {
+                    Logger.Log.UnableToGetMemoryPressureLevel(

```

Comment:
why would you ever not be able to get memory pressure level?

Reply:
I hope we won't - if the backing sysctl key or implementation changes this at least emits an error so we don't fly blind.